test/oauth-server/service/token.go

```diff
@@ -86,7 +159,9 @@ func generateIDToken(clientID string, lifeTime time.Duration, host, nonce string
 }
 
 func (requestHandler *RequestHandler) tokenOptions(w http.ResponseWriter, r *http.Request) {
-	jsonResponseWriter(w, r)
+	if err := jsonResponseWriter(w, r); err != nil {
+		writeError(w, err, http.StatusBadRequest)
```

Comment:
when w.Write fails you can only log error. it is not recommended to call twice w.WriteHeader